syntax = "proto2";

package fast_semantic_segmentation.protos;

message InputReader {
    optional bool shuffle = 1 [default=true];
    optional uint32 queue_capacity = 2 [default=200];
    optional int32 prefetch_queue_capacity = 3 [default=5];
    optional uint32 num_readers = 4 [default=32];
    repeated TFRecordInputReader tf_record_input_reader = 5;
    optional bool equal_mix = 6 [default=false];
    repeated uint32 ignore_label = 11;
}

message TFRecordInputReader {
    required string input_path = 1;
    required uint32 num_examples = 2;
    required uint32 height = 3;
    required uint32 width = 4;
    optional uint32 rheight = 5;
    optional uint32 rwidth = 6;
    optional uint32 repetitions = 7 [default=1];
}